contrib/database/sql/conn.go

```diff
@@ -203,5 +203,8 @@ func (tp *traceParams) tryTrace(ctx context.Context, qtype queryType, query stri
 			span.SetTag(k, v)
 		}
 	}
-	span.Finish(tracer.WithError(err))
+	defer span.Finish()
+	if err != nil && tp.cfg.errCheck != nil && tp.cfg.errCheck(err) {
+		span.SetTag(ext.Error, err)
+	}
```

Comment:
I'm not sure the introduced `defer` is justified. Why not simply:
```suggestion
	if err != nil && tp.cfg.errCheck != nil && tp.cfg.errCheck(err) {
		span.SetTag(ext.Error, err)
	}
	span.Finish()
```

Reply:
right, it's simple !